src/server/controllers/receipt.controller.js

```diff
@@ -53,7 +53,7 @@ export const singlePaymentReceipt = async (req, res) => {
   const paymentCode = req.params.payment_code;
   try {
     const penalty = await penaltyService.getByPaymentCode(paymentCode);
-    const paymentId = `${penalty.formattedReference}_${penalty.type}`;
```

Comment:
Does this change play nicely with all penalty types? If I recall, `formattedReference` is required to properly display IM references.

Reply:
I incorrectly used `formattedReference` when adding the single penalty receipt which formed an incorrect payment id for immobilisations. `paymentId` is only used to retrieve the payment data from the payment service.